src/ethereum/frontier/vm/memory.py

```diff
@@ -0,0 +1,112 @@
+"""
+Ethereum Virtual Machine (EVM) Memory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+EVM memory operations.
+"""
+from ethereum.frontier.vm import Evm
+from ethereum.utils import ceil32
+
+from ...base_types import U256, Bytes
+from .gas import calculate_memory_gas_cost, subtract_gas
+
+
+def extend_memory_and_subtract_gas(
```

Comment:
I also feel that we can get away with only 2 parameters to this function being the following:

- evm: Evm
- memory_size: U256

But this goes against how the yellowpaper defines the expansion. @SamWilsn @lightclient let me know your thoughts on this. Thanks.

Reply:
for the sake of consistency, I prefer to keep the pop statements as is i.e inside the method that implements the Opcode.